_rules/table-header-cells-refer-to-data-cells-d0f69e.md

```diff
@@ -0,0 +1,247 @@
+---
+id: d0f69e
+name: All table header cells have assigned data cells
+rule_type: atomic
+description: |
+  This rule checks that each table header has assigned data cells in a table element.
+accessibility_requirements:
+  wcag20:1.3.1: # Info and Relationships (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+acknowledgements:
+  authors:
+    - Jey Nandakumar
+    - Audrey Maniez
+---
+
+## Applicability
+
+The rule applies to any HTML element that has the [semantic role](#semantic-role) of [rowheader](https://www.w3.org/TR/wai-aria-1.1/#rowheader) or [columnheader](https://www.w3.org/TR/wai-aria-1.1/#columnheader) and is a [descendant](https://dom.spec.whatwg.org/#concept-tree-descendant) in the [flat tree](https://drafts.csswg.org/css-scoping/#flat-tree) of an element that:
+
+- has a [semantic role](#semantic-role) of either [table](https://www.w3.org/TR/wai-aria-1.1/#table) or [grid](https://www.w3.org/TR/wai-aria-1.1/#grid); and
+- is [visible](#visible); and
+- is [included in the accessibility tree](#included-in-the-accessibility-tree)
+
+## Expectation
+
+Each target element is [assigned](https://html.spec.whatwg.org/multipage/tables.html#header-and-data-cell-semantics) to at least one element with a [semantic role](#semantic-role) of [cell](https://www.w3.org/TR/wai-aria-1.1/#cell) or [gridcell](https://www.w3.org/TR/wai-aria-1.1/#gridcell), that is a [descendant](https://dom.spec.whatwg.org/#concept-tree-descendant) in the [flat tree](https://drafts.csswg.org/css-scoping/#flat-tree) of the same [table](https://www.w3.org/TR/wai-aria-1.1/#table) or [grid](https://www.w3.org/TR/wai-aria-1.1/#grid) element, as that of the target element.
+
+## Assumptions
+
+_There are currently no assumptions._
```

Comment:
There is. You're making a rather substantial assumption that all table headers have a relationship with some other cell in the table, and that stuff that's marked up as a table header actually is supposed to be a table header.

Reply:
Totally agree on the first part. I'd precise it saying that we assume that header cells have a "**visual** relationship" or "relationship **conveyed through presentation**", or something like that, to pinpoint it.

I agree also with the second part. As far as I can see, `th` and `td` are rendered in the same way (notably, https://html.spec.whatwg.org/multipage/rendering.html#tables-2 (which is **not normative**) doesn't make a difference), so what gives a cell (`th` or `td`) some relationship "conveyed through presentation" is the fact that it's at the start of a column or row, ie that it is supposed to be a table header…